smart_card_connector_app/src/background.js

```diff
@@ -109,11 +109,29 @@ function openWindow() {
       'readerTrackerSubscriber':
           readerTracker.addOnUpdateListener.bind(readerTracker),
       'readerTrackerUnsubscriber':
-          readerTracker.removeOnUpdateListener.bind(readerTracker)};
+          readerTracker.removeOnUpdateListener.bind(readerTracker),
+      'removeApp':
+          removeApp};
   GSC.PopupWindow.Server.createWindow(WINDOW_URL, WINDOW_OPTIONS, data);
 }
 
 /**
+ * Remove connected app (disconnect it and remove stored permission).
+ * @param {string} clientExtensionId
+ */
+function removeApp(clientExtensionId) {
+  // TODO(isandrk): Make the deeper code do the bulk of work and handling of
+  // dialog stuff, then return the result as promise and remove channel if
+  // neccessary.
+  var checkerInstance =
+      GSC.PcscLiteServerClientsManagement.ClientHandler.getPermissionsChecker();
+  checkerInstance.removeAppPermission(clientExtensionId);
```

Comment:
If the permission was forced by an admin policy (see the PermissionsChecking.ManagedRegistry class), then removal of the permission shouldn't be allowed.

First of all, this situation should be probably guarded from the UI side - i.e. by displaying your "x" button as grayed out.

But, second, this should also be checked here in the code to protect from the possible race conditions.